sw/device/tests/sim_dv/ast_clk_rst_inputs.c

```diff
@@ -0,0 +1,577 @@
+void adc_setup(bool first_adc_setup) {
+  // Enable adc interrupts.
+  CHECK_DIF_OK(dif_adc_ctrl_irq_set_enabled(&adc_ctrl, kDifAdcCtrlIrqMatchDone,
+                                            kDifToggleEnabled));
+
+  uint16_t channel0_filter0_max =
+      (kChannel0MaxHighByte << 8) | kChannel0MaxLowByte;
+  uint16_t channel0_filter0_min =
+      (kChannel0MinHighByte << 8) | kChannel0MinLowByte;
+  uint16_t channel1_filter0_max =
+      (kChannel1MaxHighByte << 8) | kChannel1MaxLowByte;
+  uint16_t channel1_filter0_min =
+      (kChannel1MinHighByte << 8) | kChannel1MinLowByte;
+
+  if (first_adc_setup) {
+    // Setup ADC configuration.
+    configure_adc_ctrl(&adc_ctrl);
+  } else {
+    CHECK_DIF_OK(dif_adc_ctrl_reset(&adc_ctrl));
+  }
+
+  en_plic_irqs(&plic);
+  // Setup ADC filters. There is one filter for each channel.
+  CHECK_DIF_OK(dif_adc_ctrl_configure_filter(
+      &adc_ctrl, kDifAdcCtrlChannel0,
+      (dif_adc_ctrl_filter_config_t){.filter = kDifAdcCtrlFilter0,
+                                     .generate_irq_on_match = true,
+                                     .generate_wakeup_on_match = true,
+                                     .in_range = true,
+                                     .max_voltage = channel0_filter0_max,
+                                     .min_voltage = channel0_filter0_min},
+      kDifToggleDisabled));
+  CHECK_DIF_OK(dif_adc_ctrl_configure_filter(
+      &adc_ctrl, kDifAdcCtrlChannel1,
+      (dif_adc_ctrl_filter_config_t){.filter = kDifAdcCtrlFilter0,
+                                     .generate_irq_on_match = true,
+                                     .generate_wakeup_on_match = true,
+                                     .in_range = true,
+                                     .max_voltage = channel1_filter0_max,
+                                     .min_voltage = channel1_filter0_min},
+      kDifToggleDisabled));
+
+  // enable filters.
+  CHECK_DIF_OK(dif_adc_ctrl_filter_set_enabled(
+      &adc_ctrl, kDifAdcCtrlChannel0, kDifAdcCtrlFilter0, kDifToggleEnabled));
+  CHECK_DIF_OK(dif_adc_ctrl_filter_set_enabled(
+      &adc_ctrl, kDifAdcCtrlChannel1, kDifAdcCtrlFilter0, kDifToggleEnabled));
+
+  if (first_adc_setup) {
+    CHECK_DIF_OK(dif_adc_ctrl_set_enabled(&adc_ctrl, kDifToggleEnabled));
+  }
+}
+
+void ast_enter_sleep_states_and_check_functionality(
+    dif_pwrmgr_domain_config_t pwrmgr_config,
+    dif_entropy_src_config_t entropy_src_config, uint32_t alert_idx) {
+  bool deepsleep;
+  int read_fifo_depth_val;
+  bool clock_check_status;
+  uint32_t unhealthy_fifos, errors, alerts;
+
+  const dif_edn_t edn0 = {
+      .base_addr = mmio_region_from_addr(TOP_EARLGREY_EDN0_BASE_ADDR)};
+  const dif_edn_t edn1 = {
+      .base_addr = mmio_region_from_addr(TOP_EARLGREY_EDN1_BASE_ADDR)};
+
+  if ((pwrmgr_config & (~kDifPwrmgrDomainOptionUsbClockInActivePower)) == 0) {
+    deepsleep = true;
+  } else {
+    deepsleep = false;
+  }
+
+  if (pwrmgr_testutils_is_wakeup_reason(&pwrmgr, 0)) {
```

Comment:
shoulnd't we use an enum here for 0 also?

Reply:
Done in #16729.